File: 프로그래머스/0/181865. 간단한 식 계산하기/간단한 식 계산하기.cpp
#include <algorithm>
#include <iostream>
#include <string>
using namespace std;

int solution(string binomial) {
    int answer = 0, opIndex;
    string first, second;
    if (binomial.find("+") != string::npos) {
        opIndex = binomial.find("+");
        first = binomial.substr(0, opIndex - 1), second = binomial.substr(opIndex + 2);
        return stoi(first) + stoi(second);
    } else if (binomial.find("-") != string::npos) {
        opIndex = binomial.find("-");
        first = binomial.substr(0, opIndex - 1), second = binomial.substr(opIndex + 2);
        return stoi(first) - stoi(second);
    } else if (binomial.find("*") != string::npos) {
        opIndex = binomial.find("*");
        first = binomial.substr(0, opIndex - 1), second = binomial.substr(opIndex + 2);
        return stoi(first) * stoi(second);
    } else {
        return -1;
    }
}

File: 프로그래머스/0/181865. 간단한 식 계산하기/간단한 식 계산하기.swift
import Foundation

func solution(_ binomial:String) -> Int {
    
    let arr = binomial.components(separatedBy: " ")
    
    switch arr[1] {
    case "+":
        return Int(arr[0])! + Int(arr[2])!
    case "-":
        return Int(arr[0])! - Int(arr[2])!
    case "*":
        return Int(arr[0])! * Int(arr[2])!
    default:
        return 0
    }
}


